pennylane/workflow/construct_tape.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains a function to extract a single tape from a QNode
+
+"""
+
+import pennylane as qml
+
+
+def construct_tape(qnode, level="user"):
+    """Constructs the tape for a designated stage in the transform program.
+
+    Args:
+        qnode (QNode): the qnode we want to get the tapes and post-processing for.
+        level (None, str, int, slice): Specifies which stage of the QNode's transform program to use for tape construction.
+
+            - ``None``: Uses the entire transformation pipeline.
+            - ``"top"``: Uses the top-level transformations.
+            - ``"user"``: Uses transformations up to and including user-defined operations.
+            - ``"device"``: Uses only device-level transformations.
+            - ``"gradient"``: Extracts the gradient-level tape.
+            - ``int``: Specifies the number of transforms to include from the start of the transformation program.
+            - ``slice``: Selects a subset of the transformation program.
```

Comment:
This is probably nitpicking but I wouldn't put `int` and `slice` in the same list as the above. This makes it look like `level=int` is a valid argument.